src/util/psbt/mod.rs

```diff
@@ -217,6 +221,55 @@ impl Decodable for PartiallySignedTransaction {
     }
 }
 
+// TODO: Move this struct into blockdata, since it is used in signature serialization inside
+//       sigScript and witness
+/// Partial signature, as used in [`PartiallySignedTransaction`], sigScript and witness fields in
+/// [`crate::TxIn`]
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct PartialSignature {
```

Comment:
nit(don't fix unless there is some other thing): in the near future we will have a separate field for Schnorr sigs, so this can be better named as `PartialEcdsaSig` or `PartialEcdsaSignature`.

Reply:
For taproot we will need a different structure - see #671

It has a different serialization logic.